ocaml/utils/import_info.ml

```diff
@@ -14,53 +14,65 @@
 
 module CU = Compilation_unit
 
-type t =
-  | Normal of CU.t * Digest.t
-  | Normal_no_crc of CU.t
-  | Other of CU.Name.t * (CU.t * Digest.t) option
+module Intf = struct
+  type t =
+    | Normal of CU.Name.t * Digest.t (* Unpacked, so compilation unit = name *)
+    | Name_only of CU.Name.t
+    | Other of CU.Name.t * (CU.t * Digest.t) option
 
-(* CR xclerc: Maybe introduce Other_no_crc to flatten the option *)
+  let create cu_name ~crc_with_unit =
+    match crc_with_unit with
+    | None -> Other (cu_name, None)
+    | Some (cu, crc) ->
+      (* For the moment be conservative and only use the [Normal] constructor
+         when there is no pack prefix at all. *)
+      if CU.Prefix.is_empty (CU.for_pack_prefix cu)
+         && CU.Name.equal (CU.name cu) cu_name
+      then Normal (cu_name, crc)
+      else Other (cu_name, Some (cu, crc))
 
-let create cu_name ~crc_with_unit =
-  match crc_with_unit with
-  | None -> Other (cu_name, None)
-  | Some (cu, crc) ->
-    (* For the moment be conservative and only use the [Normal] constructor when
-       there is no pack prefix at all. *)
-    if CU.Prefix.is_empty (CU.for_pack_prefix cu)
-       && CU.Name.equal (CU.name cu) cu_name
-    then Normal (cu, crc)
-    else Other (cu_name, Some (cu, crc))
+  let name t =
+    match t with Normal (name, _) | Name_only name | Other (name, _) -> name
 
-let create_normal cu ~crc =
-  match crc with Some crc -> Normal (cu, crc) | None -> Normal_no_crc cu
+  let impl t =
```

Comment:
Can you clarify why the semantics of this was changed from that of the old `cu` function?

Reply:
Because an `Import_info.Impl.t` _always_ has a full compilation unit, whereas it's normal for an `Import_info.Intf.t` not to have one.